napari/utils/colormaps/colormap.py

```diff
@@ -177,43 +178,20 @@ def map(self, values) -> np.ndarray:
         """
         values = np.atleast_1d(values)
 
-        precast = _cast_labels_to_minimum_dtype_auto(
-            values, len(self.colors) - 1, self.background_value
-        )
-
-        return self._map_precast(precast)
-
-    def _map_precast(self, values) -> np.ndarray:
-        """Map *precast* values to colors.
+        if values.dtype.kind == 'f':
+            values = values.astype(np.int64)
 
-        When mapping values, we first convert them to a smaller dtype for
-        performance reasons. This conversion changes the label values,
-        even for small labels. This method is used to map values that have
-        already been converted to the smaller dtype.
-
-        Parameters
-        ----------
-        values : np.ndarray
-            Values to be mapped. They must have already been downcast using
-            `_cast_labels_to_minimum_dtype_auto`.
-
-        Returns
-        -------
-        np.ndarray of shape (N, M, 4)
-            Mapped colors.
-        """
-        mapped = self.colors[values.astype(np.int64)]
-
-        mapped[values == self.background_value] = 0
-
-        # If using selected, disable all others
-        if self.use_selection:
-            cast_selection = _cast_labels_to_minimum_dtype_auto(
-                np.array([self.selection]),
-                len(self.colors) - 1,
-                self.background_value,
+        mapped = self.colors[(values - 1) % (len(self.colors) - 1) + 1]
+        if background_is_zero:
```

Comment:
I don't understand the use of background_is_zero? Why doesn't it just work to set the background value?

Reply:
The problem is with auto color mode for 32 and 64 bytes types. Where I decided to map the background to 0 in `_zero_preserving_modulo` (to avoid collision with other data points). But colormap itself does not know this, and cannot deduce this from input data.

Reply:
Blech.

Life was so much simpler when we just let the colours alias together willy-nilly. 😂 

I'll try to think about whether there is a simpler solution... Maybe we go back to something like map/map_precast? Or add a field, `background_value_texture`? This way you can have both `background_value` and `background_value_texture`, and they are the same for uint8 and uint16 but can be different for itemsize>2 (and I think also for negative background for int8/int16?)